### ISSUE TITLE
Missed a parens replacement?

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1320]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!bash
    if (
        [ "$1" = 'git' ] && \
            { [ "$2" = 'grep' ] || \
                  [ "$2" = 'ls-files' ] ;
            }
    ) || \
        [ "$1" = 'git-find' ]
    then
        fmt='/'
    fi
        
    echo $fmt
    
```

#### Here's what shellcheck currently says:
```

```
#### Here's what I wanted or expected to see:

My original code had () instead of {}. It was suggested to replace () with {} to avoid the subshell. Why didn't I get a suggestion to replace the other set of () in the if statement?

**Comment by koalaman** [comment]
ShellCheck did indeed fail to recognize this when the (..) in question contained a nested {..} or ||/&&. This has been fixed. Thanks!

**Comment by matthewpersico** [comment]
Your welcome!
